enterprise/com/src/test/java/org/neo4j/com/storecopy/FileMoveProviderTest.java

```diff
@@ -0,0 +1,258 @@
+        writeToFile( nestedFileTwo, "This is the file contained in directory B" );
+
+        // and
+        File targetFileOne = new File( targetParent, "A/file.txt" );
+        File targetFileTwo = new File( targetParent, "B/file.txt" );
+
+        // when
+        subject.traverseGenerateMoveActions( sourceParent, sourceParent ).forEach( moveToDirectory( targetParent ) );
+
+        // then
+        assertEquals( "This is the file contained in directory A", readFromFile( targetFileOne ) );
+        assertEquals( "This is the file contained in directory B", readFromFile( targetFileTwo ) );
+    }
+
+    @Test
+    public void filesAreMovedViaPageCacheWhenNecessary() throws IOException
+    {
+        // given there is a file on the default file system
+        File parentDirectory = createDirectory( testDirectory.directory( "parent" ) );
+        File aNormalFile = createFile( new File( parentDirectory, "aNormalFile.A" ) );
+
+        // and we have an expected target directory
+        File targetDirectory = createDirectory( testDirectory.directory( "targetDirectory" ) );
+        pageCache.getCachedFileSystem().mkdirs( targetDirectory );
+
+        // and there is also a file on the block device
+        File aPageCacheFile = new File( parentDirectory, "aBlockCopyFile.B" );
+        pageCache.getCachedFileSystem().mkdirs( parentDirectory );
+        StoreChannel storeChannel = pageCache.getCachedFileSystem().create( aPageCacheFile );
+        storeChannel.write( ByteBuffer.allocate( 20 ).putChar( 'a' ).putChar( 'b' ) );
+
+        // and some of these files are handled by the page cache
+        when( fileMoveActionInformer.shouldBeManagedByPageCache( any() ) ).thenReturn( false );
+        when( fileMoveActionInformer.shouldBeManagedByPageCache( eq( aPageCacheFile.getName() ) ) ).thenReturn( true );
+
+        // when the files are copied to target location
+        List<FileMoveAction> moveActions = subject.traverseGenerateMoveActions( parentDirectory, parentDirectory )
+                .collect( Collectors.toList() );//.forEach( moveToDirectory( targetDirectory ) );
+        moveActions.forEach( moveToDirectory( targetDirectory ) );
+
+        // then some files are copied over the default file system
+        File expectedNormalCopy = new File( targetDirectory, aNormalFile.getName() );
+        assertTrue( expectedNormalCopy.exists() );
+
+        // and correct files are copied over the page cache
+        File expectedPageCacheCopy = new File( targetDirectory, aPageCacheFile.getName() );
+        assertTrue( expectedPageCacheCopy.toString(), pageCache.getCachedFileSystem().fileExists( expectedPageCacheCopy ) );
+    }
+
+    private Supplier<RuntimeException> failure()
+    {
+        return () -> new RuntimeException( "Fail" );
+    }
+
+    private List<File> safeList( File dir )
+    {
+        return Arrays.asList( Optional.ofNullable( dir ).map( File::listFiles ).orElse( new File[]{} ) );
+    }
```

Comment:
This is a copy of the same-named method you have in the product code. It makes the same mistake of not using a FileSystemAbstraction.

Reply:
This applies to most of these helper methods, it seems.

Reply:
Looks like you are not actually sing this method.